### PR TITLE
Added cloud provider specific pipeline templates, updates in the readme

### PR DESCRIPTION
**Pull request by lpuskas**

### CODE REVIEW

File: .pipeline.yml.aws.template [hunk 1]
```diff
@@ -0,0 +1,62 @@
+pipeline:
+  create_cluster:
+    image: banzaicloud/plugin-pipeline-client:0.2.0
+    cluster_name: "sfpdcluster"
+    cluster_provider: "azure"
```
Comment (baluchicken) [inline .pipeline.yml.aws.template]: I think this one should be amazon

File: .pipeline.yml.azure.template [hunk 2]
```diff
@@ -1,33 +1,37 @@
...
-# Please set up these secrets on the CI/CD ui
-    secrets: [endpoint, username, password]
- 
+    azure_resource_group: "az_resource_group"
+
+    secrets: [plugin_endpoint, plugin_username, plugin_password]
+
+  install_monitoring
     deployment_name: "banzaicloud-stable/pipeline-cluster-monitor"
     deployment_release_name: "monitor"
```
Comment (baluchicken) [inline .pipeline.yml.azure.template]: we should add value {"prometheus": {"nodeExporter": {"enabled": true } } }

File: README.md [hunk 3]
```diff
@@ -10,24 +10,23 @@ The format and the description of the data is available here
...
 
 > In short: the configuration file contains the steps the project needs to go through the workflow from provisioning the environment, building the code, running tests to being deployed and executed along with project specific variables (eg.:credentials, program arguments, etc needed to assemble the execution command - `spark-submit` in this case).
 
-The current configuration comes with the location of the dataset pointing to a publicly readable S3 bucket (Warning: This may change!). Edit the '.pipeline.yml' file and change the value of the `spark_app_args`, `--dataPath` argument to point to the proper location of your dataset:
+Depending on the chosen cloud provider we provide templates to be used to start ([.pipeline.yml.aws.template](.pipeline.yml.aws.template) and [.pipeline.yml.azure.template](.pipeline.yml.azure.template)) Make a copy of the appropriate template and name it *.pipeline.yml*
+
+The templates come the location of the dataset pointing to locations ued by us for testing (Warning: You must change these locations!). Edit the '.pipeline.yml' file and change the value of the `spark_app_args`, `--dataPath` argument to point to the proper location of your dataset:
```
Comment (baluchicken) [inline README.md]: typo in used

File: README.md [hunk 4]
```diff
@@ -10,24 +10,23 @@ The format and the description of the data is available here
...
 
 ```yml
 ...
 spark_app_args: --dataPath s3a://<your-bucket>/<your-pdi-data>.csv
```
Comment (baluchicken) [inline README.md]: Is this right? I mean wasb also can be used here.

File: .pipeline.yml.aws.template (hunk 1 above)
Comment (matyix) [reply .pipeline.yml.aws.template]: yes 👍

File: .pipeline.yml.aws.template [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    spark_kubernetes_initcontainer_docker_image: banzaicloud/spark-init:v2.2.0-k8s-1.0.197
+    spark_kubernetes_resourcestagingserver_uri: http://spark-rss:10000
+    spark_kubernetes_resourcestagingserver_internal_uri: http://spark-rss:10000
+    spark_kubernetes_shuffle_labels: "app=spark-shuffle-service,spark-version=2.2.0"
+    spark_kubernetes_authenticate_driver_serviceaccount_name: "spark"
+    spark_metrics_conf: /opt/spark/conf/metrics.properties
+    spark_eventLog_enabled: true
+    spark.eventLog.dir: "s3a://spark-k8-logs/eventLog"
+    spark_app_source: target/scala-2.11/sf-police-incidents_2.11-0.1.jar
+    spark_app_args: --dataPath s3a://lp-deps-test/data/Police_Department_Incidents.csv
```
Comment (matyix) [inline .pipeline.yml.aws.template]: Is the `org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider` scenario considered @lpuskas?

Comment (lpuskas) [reply .pipeline.yml.aws.template]: We're working on it, no results yet

**Comment by baluchicken** [comment]
LGTM

File: .pipeline.yml.aws.template (hunk 5 above)
Comment (matyix) [reply .pipeline.yml.aws.template]: ask the community :)

**Pull request merged**